bench/workgen/workgen.cpp

```diff
@@ -523,9 +545,17 @@ WorkloadRunner::start_tables_create(WT_CONNECTION *conn)
         const std::string config(
           "key_format=S,value_format=S,app_metadata=\"" + DYN_TABLE_APP_METADATA);
 
+        uint32_t num_files = get_dir_num_files(_wt_home);
         int creates = 0, retries = 0;
-        while (
-          !stopping && creates < _workload->options.create_count && retries < TABLE_MAX_RETRIES) {
+        // Make sure not to exceed the maximum number of files that can exist in the database.
+        int num_files_to_create = 0;
+        if (num_files < max_files) {
+            if (num_files + _workload->options.create_count > max_files)
```

Comment:
We can combine both the if statments of L552 and L553

Reply:
Thank you for spotting this: 47512934baf242b6ce98243698276cd139610324.